addons/sale/models/sale_order.py

```diff
@@ -600,11 +600,14 @@ def _compute_amount_to_invoice(self):
         for order in self:
             order.amount_to_invoice = order.amount_total
             for invoice in order.invoice_ids.filtered(lambda x: x.state == 'posted'):
-                invoice_amount_currency = invoice.currency_id._convert(
-                    invoice.tax_totals['amount_total'] * -invoice.direction_sign,
-                    order.currency_id,
-                    invoice.company_id,
-                    invoice.date,
+                prices = invoice.line_ids.filtered(lambda x: x.sale_line_ids.order_id.id == order.id).mapped('price_total')
+                invoice_amount_currency = sum(
+                    invoice.currency_id._convert(
+                        price * -invoice.direction_sign,
+                        order.currency_id,
+                        invoice.company_id,
+                        invoice.date,
+                    ) for price in prices
                 )
```

Comment:
```suggestion
                prices = sum(invoice.line_ids.filtered(lambda x: x.sale_line_ids.order_id.id == order.id).mapped('price_total'))
                invoice_amount_currency = invoice.currency_id._convert(
                    prices * -invoice.direction_sign,
                    order.currency_id,
                    invoice.company_id,
                    invoice.date,
                )
```
Not a big deal, but I would avoid calling `_convert()` in a loop :wink: